### PR TITLE
[INOYI-5]: Program registration paragraph does not work in Sandbox

### PR DESCRIPTION
**Pull request by AleksandrPrykhodkoDev**
Original source: https://github.com/ymcatwincities/openy/issues/1197

## Steps for review

- [ ] Login as admin
- [ ] Go to Home -> Administration -> Structure -> Paragraphs types -> Manage fields
- [ ] You will see the only one field Program registration block
- [ ] Click on the Edit button
- [ ] Everything is ok
- [ ] Errors set to Drupal logs, url - `admin/reports/dblog`

### CODE REVIEW

**Comment by bjwilkins** [comment]
Can one of the admins verify this patch? Use "o+k to test" or ''t+est this please" for manual build execution.

File: modules/custom/openy_programs_search/src/Plugin/Block/ProgramsSearchBlock.php [hunk 1]
```diff
@@ -110,9 +122,17 @@ public function blockForm($form, FormStateInterface $form_state) {
       '#title' => $this->t('Categories Config'),
     ];
 
+    try {
+      $categories = $this->storage->getCategories();
+    }
+    catch (\Exception $e) {
+      $categories = [];
+      $this->logger->error($e->getMessage());
```
Comment (podarok) [inline modules/custom/openy_programs_search/src/Plugin/Block/ProgramsSearchBlock.php:154]: Could you also add a message to the user about why the form is empty and where to look for proper settings?

**Review by podarok — DISMISSED** [review]
Looks great
Small change, please @AleksandrPrykhodkoDev

File: modules/custom/openy_programs_search/src/Plugin/Block/ProgramsSearchBlock.php [hunk 2]
```diff
@@ -93,15 +128,26 @@ public function blockForm($form, FormStateInterface $form_state) {
 
     $form = parent::blockForm($form, $form_state);
     $conf = $this->getConfiguration();
+    $request = $this->requestStack->getSchemeAndHttpHost();
+    $url = "{$request}/admin/openy/integrations/daxko/programs-search";
```
Comment (dchubar-fj) [inline modules/custom/openy_programs_search/src/Plugin/Block/ProgramsSearchBlock.php]: Please use Drupal way to build the URL

File: modules/custom/openy_programs_search/src/Plugin/Block/ProgramsSearchBlock.php [hunk 3]
```diff
@@ -93,15 +128,26 @@ public function blockForm($form, FormStateInterface $form_state) {
...
       '#title' => $this->t('Locations Config'),
     ];
 
+    try {
+      $locations = $this->storage->getLocations();
+    }
+    catch (\Exception $e) {
+      $locations = [];
+      $this->logger->error($e->getMessage());
+      $this->messenger->addError($this->t("Сan't fetch Daxco data, please re-check your settings {$url}"));
```
Comment (dchubar-fj) [inline modules/custom/openy_programs_search/src/Plugin/Block/ProgramsSearchBlock.php]: Please use t() and placeholders to insert the URL

**Review by dchubar-fj — CHANGES_REQUESTED** [review]

File: modules/custom/openy_programs_search/src/Plugin/Block/ProgramsSearchBlock.php (hunk 2 above)
Comment (dchubar-fj) [reply modules/custom/openy_programs_search/src/Plugin/Block/ProgramsSearchBlock.php]: Url::fromRoute('examples.route')

File: modules/custom/openy_programs_search/src/Plugin/Block/ProgramsSearchBlock.php (hunk 3 above)
Comment (dchubar-fj) [reply modules/custom/openy_programs_search/src/Plugin/Block/ProgramsSearchBlock.php]: not fixed

**Review by dchubar-fj — CHANGES_REQUESTED** [review]

File: modules/custom/openy_programs_search/src/Plugin/Block/ProgramsSearchBlock.php (hunk 3 above)
Comment (AleksandrPrykhodkoDev) [reply modules/custom/openy_programs_search/src/Plugin/Block/ProgramsSearchBlock.php]: fixed

**Review by podarok — APPROVED** [review]

**Pull request merged**